src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -333,20 +326,16 @@ CashCheck::doApply ()
         else
         {
             // Let flow() do the heavy lifting on a check for an IOU.
-            auto const optDeliverMin = ctx_.tx[~sfDeliverMin];
-            STAmount const flowDeliver {[&optDeliverMin] (STTx const& tx)
-            {
-                // If the check cash specified Amount deliver exactly that.
-                if (! optDeliverMin)
-                    return static_cast<STAmount>(tx[sfAmount]);
-
-                // We can't use the maximum possible currency here because
-                // there might be a gateway transfer rate to account for.
-                // Since the transfer rate cannot exceed 200%, we use 1/2
-                // maxValue for our limit.
-                return STAmount { optDeliverMin->issue(),
-                    STAmount::cMaxValue / 2, STAmount::cMaxOffset };
-            }(ctx_.tx)};
+            //
+            // Note that for DeliverMin we don't know exactly how much
+            // currency we want flow to deliver.  We can't ask for the
+            // maximum possible currency because there might be a gateway
+            // transfer rate to account for.  Since the transfer rate cannot
+            // exceed 200%, we use 1/2 maxValue as our limit.
+            STAmount const flowDeliver {optDeliverMin ?
+                STAmount { optDeliverMin->issue(),
+                    STAmount::cMaxValue / 2, STAmount::cMaxOffset } :
+                static_cast<STAmount>(ctx_.tx[sfAmount])};
```

Comment:
While this avoids the argument over lambda captures, I prefer the lambda over this code - even with the captures. You're call, of course.